go.mod

```diff
@@ -1,7 +1,7 @@
 module github.com/openshift/configuration-anomaly-detection
 
-go 1.20
-
+go 1.21
```

Comment:
Bumping this requires bumping other things in this repo. Can this PR just bump golang/x/net to 0.24, so that we can bump the rest at another time?